src/pages/ChattingRoom.tsx

```diff
@@ -0,0 +1,209 @@
+import { useState, useEffect, useRef } from "react";
+import { useRecoilState, useRecoilValue } from "recoil";
+import {
+  currentUsersState,
+  typingUserState,
+  // currentChatListState,
+} from "../recoil/recoil";
+import styled from "styled-components";
+import SmallButton from "../components/SmallButton";
+import Profile from "../components/Profile";
+import LeftChat from "../components/LeftChat";
+import RightChat from "../components/RightChat";
+import { chatInterface } from "../interfaces/interface";
+import chatsData from "../json/chatsData.json";
+import Alert from "../components/Alert";
+
+const ChattingRoom = () => {
+  const currentUsers = useRecoilValue(currentUsersState);
+  const [typingUser, setTypingUser] = useRecoilState(typingUserState);
+  const nonTypingUser = currentUsers.filter(
+    (user) => user.userId !== typingUser.userId
+  )[0];
+  const [chatList, setChatList] = useState<chatInterface[]>(
+    chatsData.chattings[0].chatList
+  ); // TODO - recoil로 변경
+  // const [chatList, setChatList] = useRecoilState(currentChatListState);
+
+  const [inputText, setInputText] = useState<string>("");
+  const isTyping: boolean = inputText.trim() ? true : false;
+
+  const [isVisibleAlert, setIsVisibleAlert] = useState<boolean>(false);
+
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const bottomDivRef = useRef<HTMLDivElement>(null);
+
+  const handleInputChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setInputText(e.target.value);
+  };
+
+  const handleEnter = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+    if (!e.shiftKey && e.key === "Enter") {
+      handleSubmit();
+    }
+  };
+
+  const handleSubmit = (e?: React.FormEvent<HTMLFormElement>) => {
+    e?.preventDefault();
+    setChatList([
+      ...chatList,
+      {
+        userId: typingUser.userId,
+        message: inputText,
+        date: String(new Date()),
+        chatId: `ex${chatList.length}`,
+      },
+    ]);
+    setInputText("");
+    textareaRef.current?.focus();
+  };
```

Comment:
`trim()`을 사용하는 방법도 있고 아래 코드처럼 `replace()`와 정규식을 이용해서 공백/줄바꿈 등을 제거할 수 있습니다!
위에서 말씀드린 `chatList` 업데이트 하기 전 `inputText`를 검증하는 코드도 아래와 같습니다.

```suggestion
 const handleSubmit = (e?: React.FormEvent<HTMLFormElement>) => {
    e?.preventDefault();

    if (inputText.replace(/\s+/g, '')) {
      setChatList([
        ...chatList,
        {
          userId: typingUser.userId,
          message: inputText,
          date: String(new Date()),
          chatId: `ex${chatList.length}`,
        },
      ]);
    }

    setInputText('');
  };
```

Reply:
`replace`, `trim` 두 가지 함수로 빈 입력값 submit을 막을 수 있다는 것..! 잊고 있던 방식인데 감사합니다 👍👍